lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -122,9 +122,13 @@ def virtual_aggregate_arel(reflection, method_name, column)
                         else
                           reflection.klass.arel_attribute(column).send(method_name)
                         end
-          query       = query.select(arel_column)
-
-          t.grouping(Arel::Nodes::SqlLiteral.new(query.to_sql))
+          query       = t.grouping(Arel::Nodes::SqlLiteral.new(query.select(arel_column).to_sql))
+ 
+          if method_name == :size
+            query
+          else
+            t.grouping(Arel::Nodes::NamedFunction.new('COALESCE', [query, Arel::Nodes::SqlLiteral.new("0")]))
```

Comment:
thinking this should be part of a `default` value clause

Reply:
What do you mean?  As in this snippet:

```
Arel::Nodes::NamedFunction.new('COALESCE', [query, Arel::Nodes::SqlLiteral.new("0")])
```

Should moved to a helper method somewhere?  If so, I think that makes sense, though, arguably could wait for another PR if you don't want to tackle that now.

Up to you.

Reply:
thinking we need to introduce a `:default => 0` for virtual attributes. if that was present, then we would use the coalesce to ensure the default value